src/LineSegment.cpp

```diff
@@ -2,154 +2,209 @@
 // Created by rolf on 15-5-19.
 //
 
-#include "../include/roboteam_utils/LineSegment.h"
 #include "../include/roboteam_utils/Line.h"
+#include "../include/roboteam_utils/LineSegment.h"
 
 namespace rtt {
-    bool LineSegment::doesIntersect(const Line &line) const {
-        if (intersects(line)) {
-            return true;
-        }
-        return false;
+LineSegment::LineSegment(const Line &line) {
+    start=line.start;
+    end=line.end;
+}
+double LineSegment::length() const {
+    return (end - start).length();
+}
+
+double LineSegment::length2() const {
+    return (end - start).length2();
+}
+
+double LineSegment::slope() const {
+    return (end.y - start.y)/(end.x - start.x);
+}
+
+bool LineSegment::isVertical() const {
+    return (end.x == start.x) && (end.y != start.y);
+}
+
+Vector2 LineSegment::direction() const {
+    return Vector2(end - start);
+}
+
+double LineSegment::intercept() const {
+    return start.y - this->slope()*start.x;
+}
+
+std::pair<double, double> LineSegment::coefficients() const {
+    return {slope(), intercept()};
+}
+
+bool LineSegment::isPoint() const {
+    return start == end;
+}
+
+bool LineSegment::isParallel(const LineSegment &line) const {
+    // check if line is vertical, otherwise check the slope
+    if (this->isVertical() || line.isVertical()) {
+        return this->isVertical() && line.isVertical();
     }
+    return this->slope() == line.slope();
+}
+bool LineSegment::isParallel(const Line &line) const {
+    // check if line is vertical, otherwise check the slope
+    if (this->isVertical() || line.isVertical()) {
+        return this->isVertical() && line.isVertical();
+    }
+    return this->slope() == line.slope();
+}
+bool LineSegment::doesIntersect(const Line &line) const {
+    if (intersects(line)) {
+        return true;
+    }
+    return false;
```

Comment:
return intersects(line)

Reply:
Compiler does not allow implicit conversions from std::optional to bool, so I cannot do this.

Reply:
return static_cast<bool>(intersects(line));

But yeah I guess this is fine then (cleaner than ^), didn't realise it was an optional